dspace-server-webapp/src/main/java/org/dspace/app/rest/projection/EmbedRelsProjection.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.projection;
+
+import java.util.Set;
+
+import org.dspace.app.rest.model.LinkRest;
+import org.dspace.app.rest.model.hateoas.HALResource;
+
+/**
+ * Projection that allows a given set of rels to be embedded.
```

Comment:
Silly question, but what are "rels"?  Can we enhance the JavaDocs here a bit...it's unclear what constitutes a valid "rel". For example, how do we know what "rels" are valid to pass in?

That question is a bit "tongue in cheek", as I'm assuming that by "rels" you mean relationships, which are actually named `_links` in HAL format.  But, it'd be good to spell that out, or even consider renaming this `EmbedLinksProjection` (since any `link` is valid to pass to `embed`, I presume).  The rename here is optional, but I the the Javadocs do need improving here.

Reply:
"Rels" is common shorthand for "link relation types" as defined by RFC8288. The Spring HATEOAS folks call them that a lot (see the javadocs for "Link"), as do a few other places in DSpace code and in the wild (see the wikipedia page for "Link relation")...so I got used to doing the same.

It doesn't hurt to be explicit here, though, particularly in class-level javadocs.

> how do we know what "rels" are valid to pass in?

That info is expressed in a few places in code, but if you're consuming the REST api, your best bet for the official list should be the rest contract. The javadocs could stand to make reference to that, too.

Reply:
I'd rather we get a bit more specific in these JavaDocs, either via an example, a link to REST Contract or more explanation. I worry that assuming "rels" is a commonly understood term to all developers is a big assumption here.